openslides/config/api.py

```diff
@@ -173,19 +173,45 @@ def get_field_names(self):
         return [variable.name for variable in self.variables if variable.form_field is not None]
 
 
+CONFIG_HTML_INPUT_TYPE_MAPPING = {
+    'text': str,
+    'number': int,
+    'checkbox': bool,
+    'radio': str}
+
+
 class ConfigVariable(object):
     """
     A simple object class to wrap new config variables. The keyword
     arguments 'name' and 'default_value' are required. The keyword
-    argument 'form_field' has to be set if the variable should appear
-    on the ConfigView. The argument 'on_change' can get a callback
-    which is called every time, the variable is changed. If the argument
-    'translatable' is set, OpenSlides is able to translate the value during
-    setup of the database if the admin uses the respective command line option.
+    arguments 'html_input_type', 'label' and 'help_text' are for rendering
+    a HTML form element. If you set 'html_input_type' to 'radio' you have
+    to provide 'choices', which  is a tuple of two-tuples containing key
+    and label of every possible choice. The keyword argument validators
+    expects an interable of validator functions. Such a function gets the
+    value and raises ValidationError if the value is invalid. The keyword
+    argument 'on_change' can get a callback which is called every time, the
+    variable is changed. If the argument 'translatable' is set, OpenSlides
+    is able to translate the value during setup of the database if the
+    admin uses the respective command line option.
     """
-    def __init__(self, name, default_value, form_field=None, on_change=None, translatable=False):
+    def __init__(self, name, default_value, html_input_type='text', label=None,
+                 help_text=None, choices=None, validators=None, on_change=None,
+                 translatable=False):
+        assert html_input_type in CONFIG_HTML_INPUT_TYPE_MAPPING, (
+            "Invalid value for config attribute html_input_type.")
+        if html_input_type == 'radio':
+            assert choices is not None, ("Either config attribute 'choices' must not be None or "
+                                         "'html_input_type' must not be 'radio'.")
+        else:
+            assert choices is None, ("Either config attribute 'choices' must be None or "
```

Comment:
Here it is something different because we have a complex structure. The values None and 'radio' are permitted in general but not in special cases.